contrib/platform/test/com/sun/jna/platform/win32/Kernel32Test.java

```diff
@@ -328,6 +328,16 @@ public void testOpenProcess() {
     	assertEquals(WinError.ERROR_ACCESS_DENIED, Kernel32.INSTANCE.GetLastError());
     }
 
+    public void testQueryFullProcessImageName() {
+    	HANDLE h = Kernel32.INSTANCE.OpenProcess(0, false,
+    			Kernel32.INSTANCE.GetCurrentProcessId());
+    	char[] path = new char[1024];
+        IntByReference lpdwSize = new IntByReference(path.length);
+        boolean b = Kernel32.INSTANCE.QueryFullProcessImageName(h, 0, path, lpdwSize);
+        assertTrue(b);
+        assertTrue(lpdwSize.getValue() > 0);
```

Comment:
Provide more information when the assertion fails:
```java
assertTrue("Failed (" + Kernel32.INSTANCE.GetLastError() + ") to query process image name", b);
```

or even better - make _Kernel32Test_ extend _AbstractWin32TestSupport_ (and add _@Test_ annotations) and then use:

```java
assertCallSucceeded("QueryFullProcessImageName", Kernel32.INSTANCE.QueryFullProcessImageName(h, 0, path, lpdwSize));
```